_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,334 @@
+Maybe you want to compare optimization proposals without having to open a text editor.
+Maybe you are just a visual person.
+Regardless of the reason, we want to provide some basic instructions for getting you started!
+
+## Prerequisites
+
+For starters we need a Strimzi-managed Kafka cluster with Cruise Control enabled.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+Since Strimzi does not support creating custom Cruise Control API user roles we need to disable HTTP basic authentication settings.
+
+We can do so by configuring Cruise Control within a `Kafka` resource like this:
+
+``` 
+# kafka-cruise-control.yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  cruiseControl:
+    config:
+      webserver.security.enable: false
+```
+
+After configuring our Strimzi-managed Kafka cluster, we can deploy and connect the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) in one of two ways:
+
+* METHOD ONE: Customizing the Cruise Control pod
+* METHOD TWO: Creating a dedicated Cruise Control UI pod
+
+## METHOD ONE: Customizing the Cruise Control pod
+
+With our Strimzi-managed Kafka cluster deployed, we can create and deploy a custom Cruise Control image which includes the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application inside of it.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+
+### Inside the custom Cruise Control container
+
+Here the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) will run alongside the Cruise Control application in the same container making its requests directly to the Cruise Control REST API.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-3.png)
+
+### Building the Cruise Control UI container
+
+We can base our custom Cruise Control image on the original Strimzi Kafka image and install the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) bits on top:
+
+```bash
+#Dockerfile
+FROM quay.io/strimzi/kafka:latest-kafka-3.2.3
+
+ENV CC_UI_VERSION=0.4.0
+ENV CC_UI_HOME=./cruise-control-ui/dist/
+
+USER root
+
+RUN mkdir -p ${CC_UI_HOME}
+
+RUN curl -LO https://github.com/linkedin/cruise-control-ui/releases/download/v${CC_UI_VERSION}/cruise-control-ui-${CC_UI_VERSION}.tar.gz; \
+    tar xvfz cruise-control-ui-${CC_UI_VERSION}.tar.gz -C ${CC_UI_HOME} --strip-components=2; \
+    rm -f cruise-control-ui-${CC_UI_VERSION}.tar.gz*; \
+    echo "dev,dev,/kafkacruisecontrol/" > "${CC_UI_HOME}"static/config.csv;
+
+USER 1001
+```
+
+Then building and pushing the image to a container registry:
+```
+# When building, tag the image with the name of a container registry
+# that is accessible by the Kubernetes cluster.
+docker build . -t <registry>/<cruise-control-with-ui-image-name>
+
+# Push the image to that container registry
+docker push <registry>/<cruise-control-with-ui-image-name>
+```
```

Comment:
I think it would be great to add some example - people who never build and pushed a container image might be confused from this. E.g.

```
docker build . -t quay.io/kyguy/cruise-control-with-ui:latest
# Push the image to that container registry
docker push quay.io/kyguy/cruise-control-with-ui:latest
```

Reply:
Updated the text, let me know what you think!